### ISSUE TITLE
Shibboleth 3 IdP

### ISSUE THREAD

**Issue by gridworkz** [OCA/server-auth#20]
Hi:

Has anyone tried server-auth with a Shibboleth IdP? I've got a functioning IdP endpoint setup and I entered all the metadata in the server-auth Odoo 11 plugin. I enabled the plugin and I do see the link to login to my IdP on the Odoo login screen. When I click on it - it goes to https://<mydomain.com>/None and I get a 404 page not found error. The link address it's trying to execute is:

https://<mydomain.com>/auth_saml/get_auth_request?pid=1

I'm assuming it can't find: https://<mydomain.com>/metadata (my SP entityID)) - Are there any nginx tweaks that need to be done to make this URI work?

I'm going to re-test my metadata at Onelogin: https://www.samltool.com/validate_xml.php

Just curious if anyone has gone down this road. If so, is there any place to start checking?

Cheers,
Dave

**Comment by bennok** [comment]
Hi, 
The same situation has raised in another attempt. 

Here's  odoo's log:
2018-06-21 15:12:13,438 4583 INFO mdl_db werkzeug: 129.194.228.160 - - [21/Jun/2018 15:12:13] "GET /web/login HTTP/1.1" 302 -
2018-06-21 15:12:13,538 4583 INFO mdl_db werkzeug: 129.194.228.160 - - [21/Jun/2018 15:12:13] "GET /en_US/web/login HTTP/1.1" 200 -
2018-06-21 15:12:14,557 4583 INFO mdl_db werkzeug: 129.194.228.160 - - [21/Jun/2018 15:12:14] "POST /website/translations HTTP/1.1" 200 -
2018-06-21 15:12:19,039 4583 ERROR mdl_db odoo.addons.auth_saml.controllers.main: SAML2: _get_auth_request() takes 2 positional arguments but 3 were given
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/addons/auth_saml/controllers/main.py", line 141, in get_auth_request
    state)
TypeError: _get_auth_request() takes 2 positional arguments but 3 were given
2018-06-21 15:12:19,041 4583 INFO mdl_db werkzeug: 129.194.228.160 - - [21/Jun/2018 15:12:19] "GET /auth_saml/get_auth_request?pid=2 HTTP/1.1" 303 -
2018-06-21 15:12:19,093 4583 INFO mdl_db werkzeug: 129.194.228.160 - - [21/Jun/2018 15:12:19] "GET /None HTTP/1.1" 302 -
2018-06-21 15:12:19,637 4583 INFO mdl_db werkzeug: 129.194.228.160 - - [21/Jun/2018 15:12:19] "GET /en_US/None HTTP/1.1" 404 -

The 2 positional arguments instead of 3 is puzzling. This would seem like a coding error. 

Thank you
Benno

**Comment by nigle-dev** [comment]
I also get this both with my idp and with the default authentic provider.

ERROR odoo odoo.addons.auth_saml.controllers.main: SAML2: _get_auth_request() takes 2 positional arguments but 3 were given

Is this a working module?

**Comment by robch2342** [comment]
Is anyone actually using auth_saml for Odoo 12.0? I would love to hear how they got it running. I'm on Ubuntu 18.04 and it just refuses to work.

I managed to get the module working, but then I discovered that the lasso python module breaks the XML parsing of lxml.etree, making it impossible to install any modules with XML files. The only reason I even managed to install auth_saml in the first place is because I installed the wrong lasso module (pip3 install lasso gives you something completely different). As soon as I fixed that  (apt install python3-lasso) it was no longer possible to install the module without commenting out the lasso imports. I think this is the first time I've actually seen Odoo segfault, so that's kind of cool I guess.

First off I solved the issue where I got redirected to /None. Turns out there's three decodes in _auth_saml_validate that need to be removed. Maybe another version of lasso returns bytes instead of strings here? Otherwise I can't see how these have ever worked, since strings don't have a decode method in python 3.

After that I got redirected to the IDP, but didn't get logged in to Odoo. I solved that by slapping a self.env.cr.commit() on the end of _auth_saml_signin or it would refuse to save the SAML tokens. No idea what's going on there. No errors or anything; just wouldn't save. Never seen anything like it before.

After that it actually worked. Then I tried installing another module and fell into a pit of despair.

So, anyone want to tell me how they got it to work? What OS are you on? How did you install lasso? Did you have to modify the code?

**Comment by gridworkz** [comment]
What errors are you getting now? We use apt to install python3-lasso.

**Comment by robch2342** [comment]
@gridworkz Are you using this module? Did you get it to work as is?

Last night I tried to install lasso using the method outlined in issue #195. It didn't solve the issue when I tried it yesterday, but now that I tried to recreate the issue, it's gone. The issue appeared in odoo.tools.convert, on this line
```etree.parse(os.path.join(config['root_path'],'import_xml.rng' )))```
where it would go down into some c library and crash, saying the import_xml.rng wasn't valid. This happened whenever I tried to install a module with XML files in it. Trying to install auth_saml when this bug was active led to a segfault. I managed to confirm it in the python3 console, and narrowed it down to being caused by the _lasso import.

After applying the fixes from my previous post it's up and running. I'm just not sure if I trust it to continue running. Also, I really don't see how the code could possibly work as is. It looks like something that might have worked in python 2, but has been unsuccessfully translated to python 3.
